content/300-guides/300-upgrade-guides/200-upgrading-to-latest/10-upgrade-to-2290.mdx

```diff
@@ -0,0 +1,194 @@
+title: Upgrading to 2.29.0
+metaDescription: 'Manual steps when upgrading to 2.29.0.'
+---
+
+<TopBlock>
+
+This page describes manual upgrade steps that you need to perform after [upgrading to Prisma 2.29.0](.) if you want to use the `namedConstraints` Preview feature.
+
+</TopBlock>
+
+## Optional: Enable the <inlinecode>namedConstraints</inlinecode> Preview feature
+
+If you choose to enable the `namedConstraints` Preview feature, choose a workflow (Prisma Migrate or Introspection) and upgrade path that suits your requirements:
+
+### Upgrade paths for Prisma Migrate users
+
+Use the following guide if your project uses Prisma Migrate. Choose the upgrade path that matches your requirements:
+
+- [I want to maintain my existing constraint and index names](#upgrade-path-1-i-want-to-maintain-my-existing-constraint-and-index-names)
+- [I want to use Prisma's default constraint and index names](#upgrade-path-2-i-want-to-use-prismas-default-constraint-and-index-names)
+
+<!-- TODO Step 0 make sure all your environments are using the same constraint names  -->
+
+#### Upgrade path 1: I want to maintain my existing constraint and index names
+
+1.  Enable the `namedConstraints` Preview:
+
+    ```prisma highlight=3;normal
+    generator client {
+      provider        = "prisma-client-js"
+      previewFeatures = ["namedConstraints"]
+    }
+
+    datasource db {
+      provider = "postgresql"
+      url      = env("DATABASE_URL")
+    }
+
+    model User {
+      id    Int    @id() @default(autoincrement())
+      name  String @unique
+      posts Post[]
+    }
+
+    model Post {
+      id          Int     @id @default(autoincrement())
+      title       String
+      authorName  String  @default("Anonymous")
+      author      User?   @relation(fields: [authorName], references: [name])
+    }
+    ```
+
+1.  Introspect your **development database** to populate the Prisma schema with constraint and index names in your underlying database _that do not match Prisma's naming convention_:
+
+    ```terminal
+    npx prisma migrate introspect
+    ```
+
+    In this example, the highlighted constraints did not conform to Prisma's default naming convention and now include the `map` attribute field:
+
+    ```prisma highlight=11;normal
+    model User {
+      id    Int    @id(map: "Custom_Constraint_Name") @default(autoincrement())
+      name  String @unique
+      posts Post[]
+    }
+
+    model Post {
+      id    Int @id @default(autoincrement())
+      title String
+      authorName String @default("Anonymous")
+      author User? @relation(fields: [authorName], references: [name], map: "Custom_Foreign_Key_Constraint")
+    }
+    ```
+
+1.  Optionally introspect your **production database** and double-check that constraint names in production and development are in sync. If you have external applications or scripts that rely on a specific constraint name in production, make sure that these constraint names are represented in the Prisma schema.
```

Comment:
This makes it a bit weird and complex. Can we somehow pull this out or make it less intimidating?

Reply:
See below.